src/psyclone/tests/dynamo0p3_stubgen_test.py

```diff
@@ -0,0 +1,800 @@
+    ''' Check that the DynScalarArgs constructor raises the expected error
+    if it encounters an unrecognised type of scalar. '''
+    from psyclone.dynamo0p3 import DynScalarArgs
+    from psyclone import dynamo0p3
+    ast = fpapi.parse(os.path.join(BASE_PATH,
+                                   "testkern_one_int_scalar.f90"),
+                      ignore_comments=False)
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    # Sabotage the scalar argument to make it have an invalid type.
+    arg = kernel.arguments.args[1]
+    arg._type = "invalid-scalar-type"
+    # Monkeypatch the list of supported scalar types to include this one
+    monkeypatch.setattr(dynamo0p3, "VALID_SCALAR_NAMES",
+                        ["gh_real", "gh_integer", "invalid-scalar-type"])
+    with pytest.raises(InternalError) as err:
+        _ = DynScalarArgs(kernel)
+    assert ("Scalar type 'invalid-scalar-type' is in VALID_SCALAR_NAMES but "
+            "not handled" in str(err))
+
+
+def test_kernel_stub_ind_dofmap_errors():
+    '''Check that we raise the expected exceptions if the wrong arguments
+    are supplied to KernelStubArgList.indirection_dofmap() '''
+    ast = fpapi.parse(os.path.join(BASE_PATH,
+                                   "testkern_one_int_scalar.f90"),
+                      ignore_comments=False)
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    # create a temporary module to add code into
+    from psyclone.f2pygen import ModuleGen
+    module = ModuleGen("module_name")
+    # Now call KernStubArgList to raise an exception
+    from psyclone.dynamo0p3 import KernStubArgList
+    create_arg_list = KernStubArgList(kernel)
+    # First call it without an argument object
+    with pytest.raises(GenerationError) as excinfo:
+        create_arg_list.indirection_dofmap("w3")
+    assert "no CMA operator supplied" in str(excinfo)
+    # Second, call it with an argument object but one that is not
+    # an operator
+    with pytest.raises(GenerationError) as excinfo:
+        create_arg_list.indirection_dofmap("w3", kernel.arguments.args[1])
+    assert ("a CMA operator (gh_columnwise_operator) must be supplied but "
+            "got") in str(excinfo)
+
+
+def test_kernstubarglist_arglist_error():
+    '''Check that we raise an exception if we call the arglist method in
+    kernstubarglist without first calling the generate method'''
+    ast = fpapi.parse(os.path.join(BASE_PATH,
+                                   "testkern_one_int_scalar.f90"),
+                      ignore_comments=False)
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    # create a temporary module to add code into
+    from psyclone.f2pygen import ModuleGen
+    module = ModuleGen("module_name")
+    # Now call KernStubArgList to raise an exception
+    from psyclone.dynamo0p3 import KernStubArgList
```

Comment:
Move this to the top as it is replicated in tests.

Reply:
Done.